tests/playwright/sanity/includes/widgets/icon.test.js

```diff
@@ -1,55 +1,73 @@
 const { test, expect } = require( '@playwright/test' );
 const WpAdminPage = require( '../../../pages/wp-admin-page.js' );
+const EditorPage = require( '../../../pages/editor-page' );
+import Content from '../../../pages/elementor-panel-tabs/content';
+import EditorSelectors from '../../../selectors/editor-selectors.js';
 
-test( 'Enable SVG fit-to-size', async ( { page }, testInfo ) => {
-	const wpAdmin = new WpAdminPage( page, testInfo );
-	await wpAdmin.enableAdvancedUploads();
-	const editor = await wpAdmin.useElementorCleanPost(),
-		iconWidget = await editor.addWidget( 'icon' ),
-		iconSelector = '.elementor-element-' + iconWidget + ' .elementor-icon';
-
-	await test.step( 'Fit Aspect hidden for Icons', async () => {
-		await page.getByRole( 'button', { name: 'Style' } ).click();
-		await expect( await page.locator( '.elementor-control-fit_to_size .elementor-switch-label' ) ).toBeHidden();
+test.describe( 'Icon and social icon widget tests', () => {
+	test.beforeAll( async ( { browser }, testInfo ) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+		const wpAdmin = new WpAdminPage( page, testInfo );
+		await wpAdmin.enableAdvancedUploads();
 	} );
 
-	await test.step( 'Act', async () => {
-		await page.getByRole( 'button', { name: 'Content' } ).click();
-		const mediaUploadControl = await page.locator( '.elementor-control-media__preview' );
-		await mediaUploadControl.hover();
-		await mediaUploadControl.waitFor();
+	test.afterAll( async ( { browser }, testInfo ) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+		const wpAdmin = new WpAdminPage( page, testInfo );
+		await wpAdmin.disableAdvancedUploads();
+	} );
 
-		await page.getByText( 'Upload SVG' ).click();
+	test( 'Enable SVG fit-to-size', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo );
+		const editor = await wpAdmin.useElementorCleanPost(),
+			iconWidget = await editor.addWidget( 'icon' ),
+			iconSelector = '.elementor-element-' + iconWidget + ' .elementor-icon';
+		const contentTab = new Content( page, testInfo );
 
-		await page.setInputFiles( 'input[type="file"]', './tests/playwright/resources/test-svg-wide.svg' );
-		await page.getByRole( 'button', { name: 'Insert Media' } ).click();
+		await test.step( 'Fit Aspect hidden for Icons', async () => {
+			await page.getByRole( 'button', { name: 'Style' } ).click();
+			await expect( page.locator( '.elementor-control-fit_to_size .elementor-switch-label' ) ).toBeHidden();
+		} );
 
-		await page.getByRole( 'button', { name: 'Style' } ).click();
-		await page.locator( '.elementor-switch-label' ).click();
-		await page.getByRole( 'spinbutton', { name: 'Size' } ).fill( '300' );
-	} );
+		await test.step( 'Act', async () => {
+			await contentTab.uploadSVG();
+			await page.getByRole( 'button', { name: 'Style' } ).click();
+			await page.locator( '.elementor-switch-label' ).click();
```

Comment:
@dmaziar I'd suggest using a locator from .elementor-control-size to make this less flaky. .elementor-switch-label was not found when I tested this.

Reply:
the problem was in too fast await contentTab.uploadSVG();.
Should be fine now.